resources/src/Matroos.Resources/Services/ConfigurationService.cs

```diff
@@ -0,0 +1,72 @@
+﻿using Matroos.Resources.Services.Interfaces;
+
+using Microsoft.Extensions.Configuration;
+
+namespace Matroos.Resources.Services;
+
+public class ConfigurationService : IConfigurationService
+{
+    /// <summary>
+    /// The base configuration.
+    /// </summary>
+    private readonly IConfiguration _baseConfiguration;
+
+    /// <summary>
+    /// The configuration pool.
+    /// </summary>
+    private readonly Dictionary<string, object> _configurationPool;
+
+    public ConfigurationService(IConfiguration configuration)
+    {
+        _baseConfiguration = configuration;
+        _configurationPool = new();
+    }
+
+    /// <inheritdoc />
+    public object? Get(string key)
+    {
+        // Try to get the variable from the configuration pool.
+        _configurationPool.TryGetValue(key, out object? fromPool);
+        if (fromPool != null)
+        {
+            return fromPool;
+        }
+
+        // Try to get the variable from the environment.
+        object? fromEnvironment = Environment.GetEnvironmentVariable(key);
+        if (fromEnvironment != null)
+        {
+            _configurationPool.Add(key, fromEnvironment);
+            return fromEnvironment;
+        }
+
+        // Try to get the variable from the base configuration service.
+        object? fromBaseConfiguration = _baseConfiguration[key];
+        if (fromBaseConfiguration != null)
+        {
+            _configurationPool.Add(key, fromBaseConfiguration);
+            return fromBaseConfiguration;
+        }
+
+        return null;
```

Comment:
Any method needs to return a valid object, or throw. If not, you force the client to check for validity of return, and then the responsibility of throwing is left to it.

Null is not typesafe either.